os/arch/arm/src/armv7-a/arm_cpugating.c

```diff
@@ -40,7 +40,22 @@
 #include "barriers.h"
 
 #ifdef CONFIG_CPU_GATING
-volatile uint32_t ulFlashPG_Flag = 0;
+static volatile uint32_t g_cpugating_flag[CONFIG_SMP_NCPUS];
+
+void up_set_gating_flag_status(uint32_t CoreID, uint32_t val)
+{
+	g_cpugating_flag[CoreID] = val;
+	ARM_DSB();
+	/* Flag already reach 0 */
+	if (!g_cpugating_flag[CoreID]) {
+		SP_SEV();
```

Comment:
can you replace it by  __asm__ __volatile__ ("sev" : : : "memory"); 
to use SP_SEV(), we need to define CONFIG_ARM_HAVE_WFE_SEV but it is not .
This was causing the hang issue